botocore/paginate.py

```diff
@@ -341,9 +345,43 @@ def _parse_starting_token(self):
                 index = next_token.get('boto_truncate_amount')
                 del next_token['boto_truncate_amount']
         except (ValueError, TypeError):
-            raise ValueError("Bad starting token: %s" % self._starting_token)
+            next_token, index = self._parse_starting_token_deprecated()
         return next_token, index
 
+    def _parse_starting_token_deprecated(self):
+        """
+        This handles parsing of old style starting tokens, and attempts to
+        coerce them into the new style.
+        """
+        log.debug("Attempting to fall back to old starting token parser.")
```

Comment:
Might be good to say what the token was that was received that started this attempt.